complex.c

```diff
@@ -562,6 +562,8 @@ nucomp_f_complex(int argc, VALUE *argv, VALUE klass)
     }
     if (!NIL_P(opts)) {
         raise = rb_opts_exception_p(opts, raise);
+    } else if (argc == 1 && CLASS_OF(a1) == rb_cComplex) {
```

Comment:
I don't think this should be exclusive with keyword arguments.

Reply:
Agreed.  I've pushed another commit that handle the options, both for Complex and BigDecimal.